test/remote/gateways/remote_nuvei_test.rb

```diff
@@ -0,0 +1,59 @@
+require 'test_helper'
+require 'timecop'
+
+class RemoteNuveiTest < Test::Unit::TestCase
+  def setup
+    @gateway = NuveiGateway.new(fixtures(:nuvei))
+
+    @amount = 100
+    @credit_card = credit_card('4761344136141390', verification_value: '999', first_name: 'Cure', last_name: 'Tester')
+    @declined_card = credit_card('4000128449498204')
+
+    @options = {
+      email: 'test@gmail.com',
+      billing_address: address.merge(name: 'Cure Tester'),
+      ip_address: '127.0.0.1'
+    }
+  end
+
+  def test_transcript_scrubbing
+    transcript = capture_transcript(@gateway) do
+      @gateway.authorize(@amount, @credit_card, @options)
+    end
+
+    @gateway.scrub(transcript)
+  end
+
+  def test_successful_session_token_generation
+    response = @gateway.send(:fetch_session_token)
+    assert_success response
+    assert_not_nil response.params[:sessionToken]
+  end
+
+  def test_successful_authorize
+    response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success response
+
+    assert_not_nil response.params[:transactionId]
+    assert_match 'SUCCESS', response.message
+    assert_match 'APPROVED', response.params['transactionStatus']
+  end
```

Comment:
💬 Comment:

Same here, we need to confirm how request with errors are handled, and also confirm that token renovation works as expected.

Reply:
I will add some failure tests